src/standalone/userJupyterServer/userServerUrlProvider.ts

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+'use strict';
+import { inject, injectable, named } from 'inversify';
+import uuid from 'uuid/v4';
+import {
+    commands,
+    Disposable,
+    Event,
+    EventEmitter,
+    Memento,
+    QuickInputButtons,
+    QuickPickItem,
+    Uri,
+    window
+} from 'vscode';
+import { JupyterConnection } from '../../kernels/jupyter/jupyterConnection';
+import { validateSelectJupyterURI } from '../../kernels/jupyter/serverSelector';
+import { IJupyterServerUri, IJupyterUriProvider, IJupyterUriProviderRegistration } from '../../kernels/jupyter/types';
+import { IExtensionSyncActivationService } from '../../platform/activation/types';
+import { IApplicationShell, IClipboard, IEncryptedStorage } from '../../platform/common/application/types';
+import { Settings } from '../../platform/common/constants';
+import {
+    GLOBAL_MEMENTO,
+    IConfigurationService,
+    IDisposable,
+    IMemento,
+    IsWebExtension
+} from '../../platform/common/types';
+import { DataScience } from '../../platform/common/utils/localize';
+import { traceError } from '../../platform/logging';
+
+export const UserJupyterServerUriListKey = 'user-jupyter-server-uri-list';
+const UserJupyterServerUriListMementoKey = '_builtin.jupyterServerUrlProvider.uriList';
+
+@injectable()
+export class UserJupyterServerUrlProvider implements IExtensionSyncActivationService, IDisposable, IJupyterUriProvider {
+    readonly id: string = '_builtin.jupyterServerUrlProvider';
+    readonly displayName: string = DataScience.UserJupyterServerUrlProviderDisplayName();
+    readonly detail: string = DataScience.UserJupyterServerUrlProviderDetail();
+    private _onDidChangeHandles = new EventEmitter<void>();
+    onDidChangeHandles: Event<void> = this._onDidChangeHandles.event;
+    private readonly _globalDisposables: IDisposable[] = [];
```

Comment:
These aren't disposed,
you'd want to pull in `IDisposableRegistry` into the ctor